src/System.Management.Automation/engine/InitialSessionState.cs

```diff
@@ -4030,11 +4030,16 @@ internal static string GetClearHostFunctionText()
             {
                 // use $RawUI so this works over remoting where there isn't a physical console
                 return @"
-$RawUI = $Host.UI.RawUI
-$RawUI.CursorPosition = @{X=0;Y=0}
-$RawUI.SetBufferContents(
-    @{Top = -1; Bottom = -1; Right = -1; Left = -1},
-    @{Character = ' '; ForegroundColor = $rawui.ForegroundColor; BackgroundColor = $rawui.BackgroundColor})
+try {
+    $RawUI = $Host.UI.RawUI
+    $RawUI.CursorPosition = @{X=0;Y=0}
+    $RawUI.SetBufferContents(
+        @{Top = -1; Bottom = -1; Right = -1; Left = -1},
+        @{Character = ' '; ForegroundColor = $rawui.ForegroundColor; BackgroundColor = $rawui.BackgroundColor})
+}
+catch {
+    # fails if there is no console at all
+}
```

Comment:
This means that some scripts will not work in pwshw. I would expect this to work. That's why i think about PTY.

Reply:
I see.  So the idea is with a PTY it will fake the console so these apis return success even though there is no actual console.  Will have to look into that.

Reply:
I'll experiment with a simple conpty host spawning pwsh

Reply:
At this point, I'll continue on prototyping conpty separately, but would like to get this in as an experimental feature for 7.1 so we can get user feedback.  Will revisit conpty in 7.2 pending user feedback.